junit-platform-console/src/main/java/org/junit/platform/console/tasks/ConsoleTestExecutor.java

```diff
@@ -180,11 +186,39 @@ private Optional<TestExecutionListener> createXmlWritingListener(PrintWriter out
 	private void printSummary(TestExecutionSummary summary, PrintWriter out) {
 		// Otherwise the failures have already been printed in detail
 		if (EnumSet.of(Details.NONE, Details.SUMMARY, Details.TREE).contains(outputOptions.getDetails())) {
+			//adding diff code here
```

Comment:
This should be done in place where we print the exceptions (as it is in #3397). Otherwise, it would be difficult to find the test that belongs to the diff in case of multiple failing tests.

Reply:
![image](https://github.com/user-attachments/assets/55ce1fe2-8f78-4a62-b72c-f778edfa4cfc)
![image](https://github.com/user-attachments/assets/08a3f4ec-7164-4950-b993-8adb8e955449)

Which part do you wish to change? The first print happens in `DefaultLaucher` and the second happens in `MutableTestExecutionSummary`. Both are in `junit-platform-laucher` module. Changes will affect all Junit API output I believe? 
I can try to work on it over the weekend if you wish. 
At the same time, can you help to find why the package documentation test case fails (Test cases can not find `diff package` once I add it to the module-info.java for documentation)

Reply:
> We also need to make sure that the diff is printed if a details mode other than `Details.NONE`, `Details.SUMMARY`, or `Details.TREE` is used. We should be able to extract the diff-printing code to a separate class and call it from `MutableTestExecutionSummary`, and the other listeners (`FlatPrintingListener` etc.).

TBH, I do not know what you meant by the "details mode." All I did so far was dive into the code, find the corresponding output location of the console, and modify it.  Is there a Javadoc or class diagram or something similar I can use to learn how the Junit5 handles output?

Reply:
> 1. add diff output function to `junit-platform-commons...util.StringUtils`

I think it should stay in junit-platform-console. Please create a separate `DiffPrinter` class or similar.

Reply:
1. `DiffPrinter` class has been added.
2. Marked the location of adding diff to `FlatPrintingListener` and `VerboseTreePrintingListener`
3. Will the diff info be generated after the stack trace of the exception?